lib/floe/workflow.rb

```diff
@@ -20,16 +20,21 @@ def wait(workflows, timeout: nil, &block)
         workflows = [workflows] if workflows.kind_of?(self)
         logger.info("checking #{workflows.count} workflows...")
 
-        run_until   = Time.now.utc + timeout if timeout.to_i > 0
-        ready       = []
-        queue       = Queue.new
-        wait_thread = Thread.new do
-          loop do
-            Runner.for_resource("docker").wait do |event, runner_context|
-              queue.push([event, runner_context])
+        run_until    = Time.now.utc + timeout if timeout.to_i > 0
+        ready        = []
+        queue        = Queue.new
+        wait_threads =
+          Runner.runners.map do |runner|
+            next unless runner.respond_to?(:wait)
```

Comment:
@kbrock if we don't have a wait thread what is going to unblock the `queue.pop`
```
            # Block until an event is raised
            event, runner_context = queue.pop
```

Reply:
@agrare That thread is only needed to handle the async nature of the `"docker"` runner. I commented out the `"docker"` thread and was able to run a workflow just fine:

```json
{
  "Comment": "Directory Listing",
  "StartAt": "a",
  "States": {
    "a":{
      "Type": "Pass",
      "Next": "b"
    },
    "b": {
      "Type": "Wait",
      "Seconds": 1,
      "Next": "c"
    },
      "c": {
        "Type": "Succeed"
    }
  }
}
```

ASIDE: was also able to call with an `awesome://` `Task` without this thread